diff -ura gz-sim-gz-sim9_9.1.0/src/systems/log/LogRecord.cc gz-sim-gz-sim9_9.1.0-modified/src/systems/log/LogRecord.cc
--- gz-sim-gz-sim9_9.1.0/src/systems/log/LogRecord.cc	2025-02-15 08:15:27.000000000 +1100
+++ gz-sim-gz-sim9_9.1.0-modified/src/systems/log/LogRecord.cc	2025-04-11 17:38:01.611703341 +1000
@@ -298,7 +298,7 @@
   if (!validSdfTopic.empty())
   {
     this->sdfPub = this->node.Advertise(validSdfTopic,
-        this->sdfMsg.GetTypeName());
+        (std::string)this->sdfMsg.GetTypeName());
   }
   else
   {
diff -ura gz-sim-gz-sim9_9.1.0/src/systems/triggered_publisher/TriggeredPublisher.cc gz-sim-gz-sim9_9.1.0-modified/src/systems/triggered_publisher/TriggeredPublisher.cc
--- gz-sim-gz-sim9_9.1.0/src/systems/triggered_publisher/TriggeredPublisher.cc	2025-02-15 08:15:27.000000000 +1100
+++ gz-sim-gz-sim9_9.1.0-modified/src/systems/triggered_publisher/TriggeredPublisher.cc	2025-04-11 17:43:38.397679852 +1000
@@ -575,7 +575,7 @@
       if (nullptr != info.msgData)
       {
         info.pub =
-            this->node.Advertise(info.topic, info.msgData->GetTypeName());
+            this->node.Advertise(info.topic, (std::string)info.msgData->GetTypeName());
         if (info.pub.Valid())
         {
           this->outputInfo.push_back(std::move(info));
